src/xmpp/iq.c

```diff
@@ -308,6 +309,21 @@ iq_disco_info_request(gchar *jid)
 }
 
 void
+iq_disco_info_request_onconnect(gchar *jid)
+{
+    xmpp_ctx_t * const ctx = connection_get_ctx();
+    char *id = create_unique_id("disco_info_onconnect");
```

Comment:
I'm calling this function multiple times: Once for each server item (component).
So I need the unique ids.